tests/commands.test.ts

```diff
@@ -0,0 +1,320 @@
+import { afterAll, afterEach, beforeAll, beforeEach, describe, expect, it, jest } from "@jest/globals";
+import { RestEndpointMethodTypes } from "@octokit/plugin-rest-endpoint-methods";
+import { config } from "dotenv";
+import { http, HttpResponse } from "msw";
+import { GitHubContext } from "../src/github/github-context";
+import { GitHubEventHandler } from "../src/github/github-event-handler";
+import { CONFIG_FULL_PATH } from "../src/github/utils/config";
+import { server } from "./__mocks__/node";
+import "./__mocks__/webhooks";
+
+jest.mock("@octokit/plugin-paginate-rest", () => ({}));
+jest.mock("@octokit/plugin-rest-endpoint-methods", () => ({}));
+jest.mock("@octokit/plugin-retry", () => ({}));
+jest.mock("@octokit/plugin-throttling", () => ({}));
+jest.mock("@octokit/auth-app", () => ({}));
+
+config({ path: ".dev.vars" });
+
+const name = "ubiquity-os-kernel";
+
+const dispatchWorkflow = jest.fn();
+jest.mock("../src/github/utils/workflow-dispatch", () => ({
+  //...(jest.requireActual("../src/github/utils/workflow-dispatch") as object),
+  getDefaultBranch: async () => "main",
+  dispatchWorkflow: dispatchWorkflow,
+}));
```

Comment:
@gentlementlegen I've been fighting with Jest for hours...do you might know why is mocking working only outside the test function. In `sdk.test.ts` I put it in the test function and it worked but not here.

Also `getDefaultBranch` returns undefined when it gets called by `issueCommentCreated` and I don't understand why

Reply:
We have moved to `ESM` so I'd suggest trying `unstable_mock` instead. As such, imports should happen after the mock, since it is hoisted.

Reply:
Update: it broke in the SDK when I moved it too. It is usually easier to fake network calls than the actual package.

Reply:
I'm not even sure if kernel tests are running with CJS or ESM because if I add experimental vm modules and unstable mock, it says that `import` must be used

Reply:
even though the module `issue-comment-created` is dynamically imported in the test function, it's still importing original `dispatchWorkflow` and not mocked

Reply:
This can happen if the module you are trying to lock is previously imported by another module first, in such case the mock won't work. I've seen such scenario within the SDK.

Reply:
It seems the issue was that jest caches imported modules and I imported `issue-comment-created` in 3 test cases but only mocked `dispatchWorkflow` in the 2nd test case, but jest cached the first import so the mock didn't work. I had to use `jest.resetModules()` to reset the cache.

Reply:
Glad you figured it out. Jest has been giving me headaches a lot lately particularly when ESM came into play.